pkg/serviceaccount/externaljwt/metrics/metrics.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"crypto/sha256"
+	"errors"
+	"sync"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const (
+	namespace = "apiserver"
+	subsystem = "externaljwt"
+)
+
+var (
+	failedKeyFetch = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Namespace:      namespace,
+			Subsystem:      subsystem,
+			Name:           "fetch_jwks_error_total",
+			Help:           "Total failed attempts at syncing supported JWKs",
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{"grpc_status_code"},
+	)
+
+	totalKeyFetch = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Namespace:      namespace,
+			Subsystem:      subsystem,
+			Name:           "fetch_jwks_request_total",
+			Help:           "Total attempts at syncing supported JWKs",
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{"status"},
+	)
```

Comment:
same question as below about a single metric for public key requests, with `status` and `code` labels

Reply:
now I have:

- `fetch_keys_request_total` with "code" label to record requests to plugin.
- `sign_request_total` with "code" label to record requests to plugin.
- `outbound_request` with "method" and "code" label and hooked onto rpc conn via ChainInterceptor to record Latencies and request counts to external-jwt-signer.